diff --git a/eden/scm/gen_version.py b/eden/scm/gen_version.py
index 0fdbff8fa58..083a3169065 100755
--- a/eden/scm/gen_version.py
+++ b/eden/scm/gen_version.py
@@ -13,7 +13,7 @@ import os
 import struct
 
 
-MAIN_VERSION = "4.4.2"
+MAIN_VERSION = "0.2"
 
 
 def main() -> None:
diff --git a/eden/scm/setup.py b/eden/scm/setup.py
index b81a818ff99..b3e9751f06e 100644
--- a/eden/scm/setup.py
+++ b/eden/scm/setup.py
@@ -405,7 +405,7 @@ def pickversion():
     # Some tools parse this number to figure out if they support this version of
     # Mercurial, so prepend with 4.4.2.
     # ex. 4.4.2_20180105_214829_58fda95a0202
-    return "_".join(["4.4.2"] + out.split())
+    return "_".join(["0.2"] + out.split())
 
 
 if not os.path.isdir(builddir):
