diff --git a/eden/scm/lib/util/compilerstubs/Cargo.toml b/eden/scm/lib/util/compilerstubs/Cargo.toml
new file mode 100644
--- /dev/null
+++ b/eden/scm/lib/util/compilerstubs/Cargo.toml
@@ -0,0 +1,9 @@
+[package]
+name = "sapling-compilerstubs"
+version = "0.1.0"
+edition = "2024"
+
+[lib]
+name = "compilerstubs"
+
+[dependencies]
diff --git a/eden/scm/lib/util/compilerstubs/src/lib.rs b/eden/scm/lib/util/compilerstubs/src/lib.rs
new file mode 100644
--- /dev/null
+++ b/eden/scm/lib/util/compilerstubs/src/lib.rs
@@ -0,0 +1,35 @@
+//! Workarounds for features missing from the public stable compiler.
+
+use std::result::Result;
+use std::sync::OnceLock;
+
+/// A OnceCell that supports getting a value if one is set, or else trying to
+/// generate the value using a FnOnce that may fail, in the style of the
+/// unstable once_cell_try feature.
+///
+/// This is provided as a trait to simplify the process of mechanically
+/// replacing uses of OnceCell::get_or_try_init in the codebase.
+pub trait GetOrTryCell<T> {
+    /// Get the cell's contents if set, or else try to generate them using the
+    /// supplied function.
+    ///
+    /// This uses a different name than OnceCell::get_or_try_init to stop the
+    /// compiler from complaining about the name clashing with a likely future
+    /// feature.
+    fn get_or_try_init_stub<F, E>(&self, f: F) -> Result<&T, E>
+    where
+        F: FnOnce() -> Result<T, E>;
+}
+
+impl<T> GetOrTryCell<T> for OnceLock<T> {
+    fn get_or_try_init_stub<F, E>(&self, f: F) -> Result<&T, E>
+    where
+        F: FnOnce() -> Result<T, E>,
+    {
+        if let Some(v) = self.get() {
+            return Ok(v);
+        }
+        let v = f()?;
+        Ok(self.get_or_init(|| v))
+    }
+}
