diff --git a/eden/scm/sapling/util.py b/eden/scm/sapling/util.py
--- a/eden/scm/sapling/util.py
+++ b/eden/scm/sapling/util.py
@@ -487,10 +487,19 @@
     return bindings.version.VERSION
 
 
+VERSION_SUFFIX_PATTERN = remod.compile(r"(?:\d+\.\d+-)?(?P<suffix>.*)")
+
+
 def versionagedays() -> int:
     """Returns approximate age in days of the current version, or 0 if not available."""
     try:
         v = version()
+
+        # Allow for the 0.2- version prefix we use for packaging.
+        m = VERSION_SUFFIX_PATTERN.match(v)
+        if m:
+            v = "foo_" + m.group("suffix")
+
         parts = remod.split("_", v)
         approxbuilddate = datetime.datetime.strptime(parts[1], "%Y%m%d")
         now = datetime.datetime.now()
