diff --git a/eden/scm/lib/backingstore/Cargo.toml b/eden/scm/lib/backingstore/Cargo.toml
--- a/eden/scm/lib/backingstore/Cargo.toml
+++ b/eden/scm/lib/backingstore/Cargo.toml
@@ -41,7 +41,7 @@
 sapling-testutil = { version = "0.1.0", path = "../util/testutil" }
 sapling-types = { version = "0.1.0", path = "../types" }
 sapling-walkdetector = { version = "0.1.0", path = "../walkdetector" }
-smallvec = { version = "1.15.2", features = ["const_generics", "impl_bincode", "serde", "specialization", "union", "write"] }
+smallvec = { version = "1.15.2", features = ["const_generics", "impl_bincode", "serde", "union", "write"] }
 tracing = { version = "0.1.41", features = ["attributes", "valuable"] }
 tracing-subscriber = { version = "0.3.23", features = ["chrono", "env-filter", "json", "local-time", "parking_lot", "registry"] }
 
diff --git a/eden/scm/lib/slex/Cargo.toml b/eden/scm/lib/slex/Cargo.toml
--- a/eden/scm/lib/slex/Cargo.toml
+++ b/eden/scm/lib/slex/Cargo.toml
@@ -24,7 +24,7 @@
 parking_lot = { version = "0.12.1", features = ["arc_lock", "send_guard"] }
 sapling-async-runtime = { version = "0.1.0", path = "../async-runtime" }
 slex-items = { version = "0.1.0", path = "items" }
-smallvec = { version = "1.15.2", features = ["const_generics", "impl_bincode", "serde", "specialization", "union", "write"] }
+smallvec = { version = "1.15.2", features = ["const_generics", "impl_bincode", "serde", "union", "write"] }
 tokio = { version = "1.52.4", features = ["full", "test-util", "tracing"] }
 
 [dev-dependencies]
