diff --git a/eden/scm/lib/commands/commands/cmdversion/src/lib.rs b/eden/scm/lib/commands/commands/cmdversion/src/lib.rs
--- a/eden/scm/lib/commands/commands/cmdversion/src/lib.rs
+++ b/eden/scm/lib/commands/commands/cmdversion/src/lib.rs
@@ -15,7 +15,11 @@
 pub fn run(ctx: ReqCtx<NoOpts>) -> Result<u8> {
     let id = identity::default();
     let io = ctx.io();
-    io.write(format!("{} {}\n", id.product_name(), ::version::VERSION))?;
+    io.write(format!(
+        "{} 0.2-{}\n",
+        id.product_name(),
+        ::version::VERSION
+    ))?;
 
     if !ctx.global_opts().quiet {
         io.write_err("(see https://sapling-scm.com/ for more information)\n")?;
diff --git a/eden/scm/sapling/util.py b/eden/scm/sapling/util.py
--- a/eden/scm/sapling/util.py
+++ b/eden/scm/sapling/util.py
@@ -490,7 +490,7 @@
 def versionagedays() -> int:
     """Returns approximate age in days of the current version, or 0 if not available."""
     try:
-        v = version()
+        v = version()[4:]
         parts = remod.split("_", v)
         approxbuilddate = datetime.datetime.strptime(parts[1], "%Y%m%d")
         now = datetime.datetime.now()
