diff --git a/eden/scm/sapling/commands/isl.py b/eden/scm/sapling/commands/isl.py
--- a/eden/scm/sapling/commands/isl.py
+++ b/eden/scm/sapling/commands/isl.py
@@ -303,6 +303,7 @@
     # find "isl-dist.tar.xz"
     isl_dist_name = "isl-dist.tar.xz"
     candidates = ui.configlist("web", "isl-dist-path") + [
+        os.path.join("/usr/lib/sapling", isl_dist_name),
         os.path.join("..", "lib", isl_dist_name),
         isl_dist_name,
     ]
